auth-service/src/test/java/com/petclinic/auth/User/AuthServiceUserEntityTests.java

```diff
@@ -76,4 +86,60 @@ void user_builder() {
         assertEquals(ID, user.getId());
         assertTrue(user.getRoles().stream().anyMatch(n -> n.getName().equals(ROLE_NAME)));
     }
+
+    @Test
+    @DisplayName("Given user with roles, get authorities")
+    void getAuthorities() {
+
+        final Role parent = new Role(-1, "parent"),
+                kid = new Role(-1, "kid", parent);
+
+        HashSet<Role> basedRoles = new HashSet<>(ROLES);
+        basedRoles.add(kid);
+
+        final User build = User.builder()
+                .roles(basedRoles)
+                .build();
+
+        final Set<String> roleNames = basedRoles.parallelStream()
+                .map(n -> format("ROLE_%s", n.getName()))
+                .collect(Collectors.toSet());
+        roleNames.add(format("ROLE_%s", parent.getName())); // Manually add parent because we are not crawling up the hierarchy
+
+        assertEquals(roleNames.size(), build.getAuthorities().size());
+
+        for (GrantedAuthority authority : build.getAuthorities()) {
+            assertTrue(roleNames.contains(authority.getAuthority()));
+        }
+    }
+
+    @Test
+    @DisplayName("Given any user, account is not expired")
+    void isAccountNonExpired() {
+        assertTrue(new User().isAccountNonExpired());
+    }
+
+    @Test
+    @DisplayName("Given any user, account is not locked")
+    void isAccountNonLocked() {
+        assertTrue(new User().isAccountNonLocked());
+    }
+
+    @Test
+    @DisplayName("Given any user, credentials is non expired")
+    void isCredentialsNonExpired() {
+        assertTrue(new User().isAccountNonExpired());
+    }
+
```

Comment:
couldn't you combine these into a single test?

Reply:
I could but if in the future they branch out to have real algorithms and implementations (instead of `return true`), it would be more work